### PR TITLE
[IMP]10721 add module check_product_qty_with_material.

### PR DESCRIPTION
**Pull request by Rona111**

### CODE REVIEW

**Comment by elicoidal** [comment]
To be cleansed (some Chinese in menu etc. no translation).
README/Header

**Comment by elicoidal** [comment]
@rona111 this one is easy to fix: the Chinese menu mainly

File: check_product_qty_with_material/wizard/check_product_qty_wizard.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                    'product_id': bom.product_id.id,
+                    'product_qty': 1 * product_qty,
+                    'childs': bom_lines,
+                    'level': 0,
+                }
+                bom_map2 = self.analyze_bom_map([bom_map2], [[bom_map2]])
+                bom_map.append(bom_map2[0])
+        return bom_map
+
+    def praepare_report(self, bom_map):
```
Comment (elicoidal) [inline check_product_qty_with_material/wizard/check_product_qty_wizard.py]: s/praepare/prepare

File: check_product_qty_with_material/wizard/check_product_qty_wizard.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+class check_product_qty_wizard(models.TransientModel):
+    _name = "check.product.qty.wizard"
+    _descript = "Check Product Qty With Mateial"
+
+    product_name = fields.Char(string='产品')
+    product_qty = fields.Float(string='数量')
+    location_id = fields.Many2one('stock.location', string='库位', default=12)
+
+    @api.one
```
Comment (elicoidal) [inline check_product_qty_with_material/wizard/check_product_qty_wizard.py]: obsolete

File: check_product_qty_with_material/wizard/check_product_qty_wizard.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class check_product_qty_wizard(models.TransientModel):
+    _name = "check.product.qty.wizard"
+    _descript = "Check Product Qty With Mateial"
+
+    product_name = fields.Char(string='产品')
```
Comment (elicoidal) [inline check_product_qty_with_material/wizard/check_product_qty_wizard.py]: chinese

File: check_product_qty_with_material/wizard/check_product_qty_wizard.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
+<openerp>
+    <data>
+        <record id="check_product_qty_wizard_form" model="ir.ui.view">
+            <field name="name">check.product.qty.wizard.form</field>
+            <field name="model">check.product.qty.wizard</field>
+            <field name="type">form</field>
+            <field name="arch" type="xml">
+                <form string="计算产品可生产数量" version="7">
```
Comment (elicoidal) [inline check_product_qty_with_material/wizard/check_product_qty_wizard.xml]: version is 7 or 8?

File: check_product_qty_with_material/README.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+===============================
+Check product qty with material
+===============================
+
+This module extends the functionality of ... to support ...
```
Comment (elicoidal) [inline check_product_qty_with_material/README.rst]: not sure what this module do: prepare a description and I will correct it.

module name: product_bom_qty_check

**Comment by noahzaozao** [comment]
@elicoidal please review this pr again

File: check_product_qty_with_material/product.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+from openerp import fields, models
+
+
+class CheckProductQtyReport(models.Model):
+    _name = "check.product.qty.report"
+    _descript = "Check Product Qty Report"
+
+    product_name = fields.Char(string='Product')
+    product_qty = fields.Float(string='Number of material')
```
Comment (elicoidal) [inline check_product_qty_with_material/product.py]: Qty of Raw materials

File: check_product_qty_with_material/product.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    stock_qty = fields.Float(string='Number of stock')
```
Comment (elicoidal) [inline check_product_qty_with_material/product.py]: Stock Qty

File: check_product_qty_with_material/product.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    can_product_qty = fields.Float(string='Number can be produced')
```
Comment (elicoidal) [inline check_product_qty_with_material/product.py]: Qty that can be produced

File: check_product_qty_with_material/product.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    diff_qty = fields.Float(string='Number of differences')
```
Comment (elicoidal) [inline check_product_qty_with_material/product.py]: Difference qty

File: check_product_qty_with_material/README.rst [hunk 10]
```diff
@@ -0,0 +1,61 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=====================
+Product Bom Qty Check
+=====================
+
+This module extends the functionality of ... to support ...
```
Comment (elicoidal) [inline check_product_qty_with_material/README.rst]: Reports to display the quantity of finished good that can be produced based on the quantity on stock of the raw materials.

File: check_product_qty_with_material/README.rst [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    and allow you to ...
+
+Installation
```
Comment (elicoidal) [inline check_product_qty_with_material/README.rst]: remove this section

File: check_product_qty_with_material/README.rst [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+ * have basic modules installed (stock, mrp)
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/Elico-Corp/odoo/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/Elico-Corp/odoo/issues/new?body=module:%20check_product_qty_with_material%0Aversion:%20{8.0}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (elicoidal) [inline check_product_qty_with_material/README.rst]: remove this line

File: check_product_qty_with_material/wizard/check_product_qty_wizard.xml [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+                    </group>
+                    <footer>
+                        <button name="apply" class="oe_highlight" string="计算" type="object"/>
+                        <button special="cancel" string="取消"/>
+                    </footer>
+                </form>
+            </field>
+        </record>
+
+        <act_window name="计算产品可生产数量" res_model="check.product.qty.wizard" view_mode="form" target="new" id="action_check_product_qty_wizard"/>
```
Comment (elicoidal) [inline check_product_qty_with_material/wizard/check_product_qty_wizard.xml]: remove Chinese

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details to be fixed. Could be done by Kevin

**Comment by noahzaozao** [comment]
@elicoidal please review this pr again

**Review by elicoidal — APPROVED** [review]

**Pull request merged**